modules/delegates.js

```diff
@@ -353,112 +353,160 @@ __private.decryptPassphrase = function(encryptedPassphrase, password) {
  * @returns {setImmediateCallback} cb, err
  * @todo Add description for the params
  */
-__private.checkDelegates = function(publicKey, votes, state, cb, tx) {
+__private.checkDelegates = function(senderPublicKey, votes, state, cb, tx) {
 	if (!Array.isArray(votes)) {
 		return setImmediate(cb, 'Votes must be an array');
 	}
 
-	modules.accounts.getAccount(
-		{ publicKey },
-		(err, account) => {
-			if (err) {
-				return setImmediate(cb, err);
-			}
+	// Converting publicKeys to addresses because module.accounts do not support fetching multiple accounts by publicKeys
+	// TODO: Use Hashmap to improve performance further.
+	const actionWithAddresses = votes.map(vote => {
```

Comment:
More readable name suggestion is `voteAddressesWithActions`

Reply:
Why?

Reply:
Its a suggestion. To me it's more readable.